driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/ChangeStreamPublisherImpl.java

```diff
@@ -112,12 +112,18 @@ private ChangeStreamPublisherImpl(
 
     @Override
     public <TDocument> Publisher<TDocument> withDocumentClass(final Class<TDocument> clazz) {
-        return new BatchCursorPublisher<TDocument>(getClientSession(), getMongoOperationPublisher().withDocumentClass(clazz)) {
+        BatchCursorPublisher<TDocument> result = new BatchCursorPublisher<TDocument>(
+                getClientSession(), getMongoOperationPublisher().withDocumentClass(clazz)) {
             @Override
             AsyncReadOperation<AsyncBatchCursor<TDocument>> asAsyncReadOperation(final int initialBatchSize) {
                 return createChangeStreamOperation(getMongoOperationPublisher().getCodecRegistry().get(clazz), initialBatchSize);
             }
         };
+        Integer batchSize = getBatchSize();
+        if (batchSize != null) {
+            result.batchSize(batchSize);
```

Comment:
It's not clear to me why batchSize is the one property that isn't propagated.  Can you explain what's going on?

Reply:
It's actually not clear to me either. That's why I asked Ross directly in another comment.